source/MapPanel.cpp

```diff
@@ -1386,10 +1398,36 @@ void MapPanel::DrawSystems()
 
 	// Draw the circles for the systems.
 	double zoom = Zoom();
+
+	const float ringFade = isStarry ? 1.5 - 1.25 * zoom : 1.;
 	for(const Node &node : nodes)
 	{
 		Point pos = zoom * (node.position + center);
-		RingShader::Draw(pos, OUTER, INNER, node.color);
+
+		// System rings fade as you zoom in if starry map is enabled.
+		RingShader::Draw(pos, OUTER, INNER, node.color.Additive(max(ringFade,
+			node.mapIcon.empty() ? .9f : 0)));
+
+		if(isStarry)
+		{
+			// Ensures every multiple-star system has a characteristic, deterministic rotation.
+			float starAngle = node.name.length() + node.position.Length();
+			float spin = 4 * acos(0.) / node.mapIcon.size();
```

Comment:
What is `acos(0.)` doing here?
I'm guessing this is calculating the angular size of n equal segments around a circle, where n is the number of icons. I think a name like `spacing` would far more clearly indicate this, and simply dividing `2 * pi` by the number of items would much more clearly show what you are calculating here, rather than this obfuscation.

What happens if `node.mapIcon` is empty? The code drawing the ring above checks if it is empty, so I assume it is not guaranteed to have at least one element, and if it is, that other code should be updated to reflect that rather than checking a condition that will always be true.

Reply:
When calculating the alpha value for the ring, you use a different intermediate value depending on if `node.mapIcons` is empty or not:
```c++
			const float alpha = max(ringFade, node.mapIcons.empty() ? .9f : 0.f);
```
This suggests that it is possible for `mapIcons` to be empty. If it is in fact not possible for it to be empty at this point (I don't think this is the case), this check should not be present.

In this line, you do:
```c++
			float spin = 4 * acos(0.) / node.mapIcon.size();
```
Now:
```c++
			Angle spacing = 360. / node.mapIcons.size();
```
I don't see anything between the other point and this one that would ensure that `node.mapIcons` is not empty, so it probably can still be empty (we can test by removing the `icon` entries for some `star` definitions) which will result in `node.mapIcons.size()` being zero, and dividing non-zero things by zero is generally not advisable.

Reply:
> I've tested it with empty mapIcons and everything seems to work fine.

It's still dividing by zero. The result is never used in that case, so that part doesn't matter, but dividing by zero is undefined behaviour, which could be anything from just returning whatever value was in the register to crashing the whole application, depends on the implementation. Just because you can't see any problems with whatever compiler you've tested it with doesn't mean it's something we should be doing.